### ISSUE TITLE
[GLPI 10.0-RC1]+[AgentGlPI_Nightly build] : Display bug after updating the agent on a computer

### ISSUE THREAD

**Issue by steph461** [glpi-project/glpi#10676]
### Bug reporting acknowledgment

Yes, I read it

### Describe the bug

================================================================================
1) Update my agent on glpi agent (Fusion====>agent glpi)
I can no longer perform discovery from GLPI

Before maj
![before](https://user-images.githubusercontent.com/68054991/154266203-c682a173-adea-42b4-b4ce-818a4942b765.jpg)

After maj

![After_maj](https://user-images.githubusercontent.com/68054991/154266236-a194db17-d5a7-42e7-8ac8-f756f8102f66.jpg)

```shell
[2022-02-16 13:07:49] glpisqllog.ERROR: DBmysql::query() in /var/www/html/glpi10beta/src/DBmysql.php line 360
  *** MySQL query error:
  SQL: INSERT INTO `glpi_agents` (`deviceid`, `name`, `last_contact`, `useragent`, `agenttypes_id`, `itemtype`, `version`, `items_id`, `threads_networkdiscovery`, `threads_networkinventory`, `timeout_networkdiscovery`, `timeout_networkinventory`) VALUES ('STIMP.LabmAlphabio.marseille.siege-2022-02-16-12-47-12', 'STIMP.LabmAlphabio.marseille.siege-2022-02-16-12-47-12', '2022-02-16 13:07:49', 'GLPI-Agent_v1.2-git70294ae7', NULL, 'Computer', '1.2-git70294ae7', '0', '0', '0', '0', '0')
  Error: Column 'agenttypes_id' cannot be null
  Backtrace :
  src/DBmysql.php:1277                               
  src/CommonDBTM.php:653                             DBmysql->insert()
  src/CommonDBTM.php:1245                            CommonDBTM->addToDB()
  src/Agent.php:387                                  CommonDBTM->add()
  src/Inventory/Inventory.php:302                    Agent->handleAgent()
  src/Inventory/Request.php:360                      Glpi\Inventory\Inventory->doInventory()
  src/Inventory/Request.php:83                       Glpi\Inventory\Request->inventory()
  src/Agent/Communication/AbstractRequest.php:305    Glpi\Inventory\Request->handleAction()
  src/Agent/Communication/AbstractRequest.php:219    Glpi\Agent\Communication\AbstractRequest->handleJSONRequest()
  front/inventory.php:67                             Glpi\Agent\Communication\AbstractRequest->handleRequest()
  {"user":"inventory@srv-glpi"} 
[2022-02-16 13:07:49] glpisqllog.ERROR: DBmysql::query() in /var/www/html/glpi10beta/src/DBmysql.php line 360
  *** MySQL query error:
  SQL: INSERT INTO `glpi_rulematchedlogs` (`date`, `rules_id`, `agents_id`, `method`, `items_id`, `itemtype`) VALUES ('2022-02-16 13:07:49', '751', NULL, 'inventory', '542', 'Computer')
  Error: Column 'agents_id' cannot be null
  Backtrace :
  src/DBmysql.php:1277                               
  src/CommonDBTM.php:653                             DBmysql->insert()
  src/CommonDBTM.php:1245                            CommonDBTM->addToDB()
  src/RuleImportAsset.php:939                        CommonDBTM->add()
  src/Rule.php:1502                                  RuleImportAsset->executeActions()
  src/RuleCollection.php:1587                        Rule->process()
  src/Inventory/Asset/MainAsset.php:515              RuleCollection->processAllRules()
  src/Inventory/Inventory.php:681                    Glpi\Inventory\Asset\MainAsset->handle()
  src/Inventory/Inventory.php:328                    Glpi\Inventory\Inventory->handleItem()
  src/Inventory/Request.php:360                      Glpi\Inventory\Inventory->doInventory()
  src/Inventory/Request.php:83                       Glpi\Inventory\Request->inventory()
  src/Agent/Communication/AbstractRequest.php:305    Glpi\Inventory\Request->handleAction()
  src/Agent/Communication/AbstractRequest.php:219    Glpi\Agent\Communication\AbstractRequest->handleJSONRequest()
  front/inventory.php:67                             Glpi\Agent\Communication\AbstractRequest->handleRequest()
  {"user":"inventory@srv-glpi","mem_usage":"0.418\", 8.57Mio)"} 
[2022-02-16 13:07:50] glpisqllog.ERROR: DBmysql::query() in /var/www/html/glpi10beta/src/DBmysql.php line 360
  *** MySQL query error:
  SQL: INSERT INTO `glpi_rulematchedlogs` (`date`, `rules_id`, `items_id`, `itemtype`, `agents_id`, `method`) VALUES ('2022-02-16 13:07:50', '751', '542', 'Computer', NULL, 'inventory')
  Error: Column 'agents_id' cannot be null
  Backtrace :
  src/DBmysql.php:1277                               
  src/CommonDBTM.php:653                             DBmysql->insert()
  src/CommonDBTM.php:1245                            CommonDBTM->addToDB()
  src/Inventory/Asset/MainAsset.php:693              CommonDBTM->add()
  src/RuleImportAsset.php:942                        Glpi\Inventory\Asset\MainAsset->rulepassed()
  src/Rule.php:1502                                  RuleImportAsset->executeActions()
  src/RuleCollection.php:1587                        Rule->process()
  src/Inventory/Asset/MainAsset.php:515              RuleCollection->processAllRules()
  src/Inventory/Inventory.php:681                    Glpi\Inventory\Asset\MainAsset->handle()
  src/Inventory/Inventory.php:328                    Glpi\Inventory\Inventory->handleItem()
  src/Inventory/Request.php:360                      Glpi\Inventory\Inventory->doInventory()
  src/Inventory/Request.php:83                       Glpi\Inventory\Request->inventory()
  src/Agent/Communication/AbstractRequest.php:305    Glpi\Inventory\Request->handleAction()
  src/Agent/Communication/AbstractRequest.php:219    Glpi\Agent\Communication\AbstractRequest->handleJSONRequest()
  front/inventory.php:67                             Glpi\Agent\Communication\AbstractRequest->handleRequest()
  {"user":"inventory@srv-glpi","mem_usage":"1.082\", 11.67Mio)"} 
```

==============================================================================
2) If iI delete my computer of GLPI and I create the computer, i have same issue 

[sql-errors.log](https://github.com/glpi-project/glpi-agent/files/8079982/sql-errors.log)

[php-errors.log](https://github.com/glpi-project/glpi-agent/files/8079981/php-errors.log)

=============================================================================
3) If I try to update the agent a second time, the modification of the date is no longer done

/front/computer.form.php?id=544
![image](https://user-images.githubusercontent.com/68054991/154271478-10b8def6-2da4-413f-9fd4-d77b8da58bfe.png)
![image](https://user-images.githubusercontent.com/68054991/154271588-91b8448e-cc5a-48c9-bad4-d57d92cb4752.png)
MAJ KO
[php-errors.log](https://github.com/glpi-project/glpi-agent/files/8080123/php-errors.log)
[sql-errors.log](https://github.com/glpi-project/glpi-agent/files/8080124/sql-errors.log)

=============================================================================

### To reproduce

1) http://localhost:62354 for reproduce issue

### Expected behavior

.

### Operating system

Linux

### GLPI Agent version

Nightly build (git version in additional context below)

### GLPI version

10.0.0-rc1

### GLPIInventory plugin or FusionInventory for GLPI plugin version

1.0.0-rc4

**Comment by steph461** [comment]
for more information : 
![image](https://user-images.githubusercontent.com/68054991/154290084-8f82dd5c-661e-4041-b8ec-5b1dea9047fb.png)
the agent-GLPI of the computer station is not created in the listing.

**Comment by orthagh** [comment]
I think this have been fixed by #10706 (reopen if not)
